cmd/manager/main.go

```diff
@@ -142,28 +154,12 @@ func main() {
 	logger.Info("Starting the Cmd.")
 
 	// Start the Cmd
-	logging.Fatal(mgr.Start(stop), "error during manager execution.")
-}
-
-func newManager(ctx context.Context, restCfg *rest.Config, scopedNamespace string, userProjectOverride bool, billingProject string) (manager.Manager, error) {
-	krmtotf.SetUserAgentForTerraformProvider()
-	controllersCfg := kccmanager.Config{
-		ManagerOptions: manager.Options{
-			Cache: cache.Options{
-				DefaultNamespaces: map[string]cache.Config{
-					scopedNamespace: {},
-				},
-			},
-		},
-	}
-
-	controllersCfg.UserProjectOverride = userProjectOverride
-	controllersCfg.BillingProject = billingProject
-	// TODO(b/320784855): StateIntoSpecDefaultValue and StateIntoSpecUserOverride values should come from the flags.
-	controllersCfg.StateIntoSpecDefaultValue = stateintospec.StateIntoSpecDefaultValueV1Beta1
-	mgr, err := kccmanager.New(ctx, restCfg, controllersCfg)
-	if err != nil {
-		return nil, fmt.Errorf("error creating manager: %w", err)
+	mgrErr := mgr.Start(stop)
+	if mgrErr != nil {
+		logging.Fatal(mgrErr, "error during manager execution.")
+	} else {
+		// err is nil
+		// todo acpana add more defense in depth here
+		logging.ExitInfo("might've lost leader election")
```

Comment:
walter had suggested adding some defense in depth here where we as the kccmanager process check our leadership status and bail out if we are no longer a leader.

however, AFAICT, we get that guarantee for free if mgr.Start returns, meaning we are no longer a leader if err is nil.

Reply:
I think we should be able to listen for a cancel on the context used for leader election and exist when it is called.